src/vm/object_model.rs

```diff
@@ -1,67 +1,58 @@
 use atomic::Ordering;
 
+use self::specs::*;
 use crate::plan::AllocationSemantics;
 use crate::plan::CopyContext;
-use crate::util::metadata::{header_metadata::HeaderMetadataSpec, MetadataSpec};
+use crate::util::metadata::header_metadata::HeaderMetadataSpec;
 use crate::util::{Address, ObjectReference};
 use crate::vm::VMBinding;
 
 /// VM-specific methods for object model.
 ///
-/// MMTk does not require but recommands using in-header per-object metadata for better performance.
-/// MMTk requires VMs to announce whether they can provide certain per-object metadata in object headers by overriding the metadata related constants in the ObjectModel trait.
+/// This trait includes 3 parts:
```

Comment:
Probably enumerate the 3 parts, instead of using *.
I think that's visually easier to follow

Reply:
Sounds good.